.github/workflows/main.yml

```diff
@@ -16,25 +16,24 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        include:
-          - container: 'ubuntu:22.04'
-            test: 'yes'
-          - container: 'ubuntu:24.04'
-            test: 'yes'
-          - container: 'archlinux:base-devel'
-            test: 'yes'
-          - container: 'fedora:latest'
-            test: 'yes'
-          - container: 'alpine:latest'
-            test: 'no'
-          - container: 'debian:unstable'
-            test: 'yes'
+        container:
+          - 'ubuntu:22.04'
+          - 'ubuntu:24.04'
+          - 'archlinux:base-devel'
+          - 'fedora:latest'
+          - 'alpine:latest'
+          - 'debian:unstable'
+        build:
+          - 'meson'
+          - 'autotools'
 
     container:
       image: ${{ matrix.container }}
 
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          sparse-checkout: .github
```

Comment:
humn... apparently 2 different things in this commit:  one is about building with meson, another is about using git to clone the repo.

Reply:
do we need it as prep commit? is it faster?